src/control/control.cpp

```diff
@@ -43,32 +43,25 @@ QWeakPointer<ControlDoublePrivate> s_pDefaultCO;
 
 // TODO: re-evaluate whether this is needed.
 ControlDoublePrivate::ControlDoublePrivate()
-        : ControlDoublePrivate({}, nullptr, true, false, false, kDefaultValue, true){};
-
-ControlDoublePrivate::ControlDoublePrivate(
-        const ConfigKey& key,
-        ControlObject* pCreatorCO,
-        bool bIgnoreNops,
-        bool bTrack,
-        bool bPersist,
-        double defaultValue,
-        bool confirmRequired = false)
-        : m_key(key),
-          m_pBehavior(nullptr),
-          m_name(QString()),
-          m_description(QString()),
-          m_value(defaultValue),
-          m_defaultValue(defaultValue),
-          m_pCreatorCO(pCreatorCO),
-          m_trackingKey(bTrack ? statTrackingKey.arg(key.group, key.item) : QString()),
-          m_confirmRequired(confirmRequired),
-          m_bPersistInConfiguration(bPersist),
-          m_bIgnoreNops(bIgnoreNops),
-          m_kbdRepeatable(false) {
-    if (bPersist) {
+        : ControlDoublePrivate({{}, true}){};
+
+ControlDoublePrivate::ControlDoublePrivate(const ParametersWithConfirm& params)
+        : m_key(params.key),
+          m_pBehavior(params.behavior),
+          m_name(params.name),
+          m_description(params.description),
+          m_value(params.defaultValue),
```

Comment:
This is re-initialized below. Maybe add a free helper function that does it on one go.

Reply:
are you referring to the `m_value.setValue`? What value should we initialize with if the config is null? just `defaultValue`?

Reply:
done in fixup